Fix detection of libiconv so that we actually use it instead of private
macOS functions.
https://github.com/jca02266/lha/issues/16
https://github.com/haikuports/haikuports/blob/master/app-arch/lha/patches/lha-1.14i.ac20191005.patchset
--- configure.ac.orig
+++ configure.ac
@@ -106,7 +106,7 @@ AC_CHECK_FUNCS(strchr strdup memcpy memset memmove strcasecmp)
 AC_CHECK_FUNCS(mktime timelocal tzset ftime gettimeofday utime)
 AC_CHECK_FUNCS(ftruncate chsize lchown mkstemp ftello)
 AC_CHECK_FUNCS(getpwuid getgrgid getpwnam getgrnam)
-AC_CHECK_FUNCS(iconv basename)
+AC_CHECK_FUNCS(libiconv iconv basename)
 AC_REPLACE_FUNCS(vsnprintf fnmatch)
 
 if test $ac_header_dirent = no; then
--- src/header.c.orig	2019-10-04 22:58:55.000000000 -0500
+++ src/header.c	2020-12-06 23:31:25.000000000 -0600
@@ -1907,7 +1907,7 @@
 
 #if MULTIBYTE_FILENAME
 
-#if defined(__APPLE__) && !defined(HAVE_ICONV)  /* Added by Hiroto Sakai */
+#if defined(__APPLE__) && !(defined(HAVE_ICONV) || defined(HAVE_LIBICONV))  /* Added by Hiroto Sakai */
 
 #include <CoreFoundation/CFString.h>
 #include <CoreFoundation/CFStringEncodingExt.h>
@@ -2000,7 +2000,7 @@
     return cfResult;
 }
 
-#elif HAVE_ICONV
+#elif defined(HAVE_ICONV) || defined(HAVE_LIBICONV)
 #include <iconv.h>
 
 static int
@@ -2045,7 +2045,7 @@
 sjis_to_utf8(char *dst, const char *src, size_t dstsize)
 {
 #if defined(__APPLE__)
-#if !defined(HAVE_ICONV)
+#if !(defined(HAVE_ICONV) || defined(HAVE_LIBICONV))
   dst[0] = '\0';
   if (ConvertEncodingToUTF8(src, dst, dstsize,
                             kCFStringEncodingDOSJapanese,
@@ -2055,7 +2055,7 @@
   if (ConvertEncodingByIconv(src, dst, dstsize, "SJIS", "UTF-8-MAC") != -1)
       return dst;
 #endif
-#elif HAVE_ICONV
+#elif defined(HAVE_ICONV) || defined(HAVE_LIBICONV)
   if (ConvertEncodingByIconv(src, dst, dstsize, "SJIS", "UTF-8") != -1)
       return dst;
 #else
@@ -2071,7 +2071,7 @@
 utf8_to_sjis(char *dst, const char *src, size_t dstsize)
 {
 #if defined(__APPLE__)
-#if !defined(HAVE_ICONV)
+#if !(defined(HAVE_ICONV) || defined(HAVE_LIBICONV))
   int srclen;
 
   dst[0] = '\0';
@@ -2084,7 +2084,7 @@
   if (ConvertEncodingByIconv(src, dst, dstsize, "UTF-8-MAC", "SJIS") != -1)
       return dst;
 #endif
-#elif HAVE_ICONV
+#elif defined(HAVE_ICONV) || defined(HAVE_LIBICONV)
   if (ConvertEncodingByIconv(src, dst, dstsize, "UTF-8", "SJIS") != -1)
       return dst;
 #else
